frontend/packages/kubevirt-plugin/src/components/modals/disk-modal/disk-modal.tsx

```diff
@@ -257,8 +257,8 @@ export const DiskModal = withHandlePromise((props: DiskModalProps) => {
         url:
           source.getDataVolumeSourceType() === DataVolumeSourceType.REGISTRY ? containerImage : url,
       })
-      .setVolumeMode(volumeMode || null)
-      .setAccessModes(accessMode ? [accessMode] : null)
+      .setVolumeMode(applySP && spVolumeMode ? spVolumeMode : volumeMode || null)
+      .setAccessModes(applySP && spAccessMode ? [spAccessMode] : accessMode ? [accessMode] : null)
```

Comment:
I'm not a huge fan of nested ternaries and definitely not of single-line nested ternaries, but I'm not going to hold this PR up over it. =)